unity-renderer/Assets/DCLServices/WearablesCatalogService/LambdasWearablesCatalogService.cs

```diff
@@ -0,0 +1,308 @@
+
+        private Service<ILambdasService> lambdasService;
+        private CancellationTokenSource serviceCts;
+        private readonly Dictionary<string, int> wearablesInUseCounters = new ();
+        private readonly Dictionary<string, LambdaResponsePagePointer<WearableResponse>> ownerWearablesPagePointers = new ();
+        private readonly Dictionary<(string, string), LambdaResponsePagePointer<WearableResponse>> thirdPartyCollectionPagePointers = new ();
+        private readonly Dictionary<string, UniTaskCompletionSource<WearableItem>> awaitingWearableTasks = new ();
+        private readonly List<string> pendingWearablesToRequest = new ();
+
+        public LambdasWearablesCatalogService(BaseDictionary<string, WearableItem> wearablesCatalog)
+        {
+            WearablesCatalog = wearablesCatalog;
+        }
+
+        public void Initialize()
+        {
+            serviceCts = serviceCts.SafeRestart();
+
+            try
+            {
+                // Check unused wearables (to be removed from our catalog) only every [TIME_TO_CHECK_FOR_UNUSED_WEARABLES] seconds
+                CheckForUnusedWearablesAsync(serviceCts.Token).Forget();
+            }
+            catch (OperationCanceledException) { }
+        }
+
+        public void Dispose()
+        {
+            serviceCts.SafeCancelAndDispose();
+            serviceCts = null;
+            Clear();
+        }
+
+        public async UniTask<IReadOnlyList<WearableItem>> RequestOwnedWearablesAsync(string userId, int pageNumber, int pageSize, bool cleanCachedPages, CancellationToken ct)
+        {
+            var createNewPointer = false;
+            if (!ownerWearablesPagePointers.TryGetValue(userId, out var pagePointer))
+            {
+                createNewPointer = true;
+            }
+            else if (cleanCachedPages)
+            {
+                pagePointer.Dispose();
+                ownerWearablesPagePointers.Remove(userId);
+                createNewPointer = true;
+            }
+
+            if (createNewPointer)
+            {
+                ownerWearablesPagePointers[userId] = pagePointer = new (
+                    WEARABLES_BY_OWNER_END_POINT.Replace("{userId}", userId),
+                    pageSize, ct, this);
+            }
+
+            var pageResponse = await pagePointer.GetPageAsync(pageNumber, ct);
+            AddWearablesToCatalog(pageResponse.response.wearables);
+
+            return pageResponse.response.wearables;
+        }
+
+        public async UniTask<IReadOnlyList<WearableItem>> RequestBaseWearablesAsync(CancellationToken ct)
+        {
+            var serviceResponse = await lambdasService.Ref.Get<WearableResponse>(
+                "",
+                WEARABLES_BY_COLLECTION_END_POINT.Replace("{collectionId}", BASE_WEARABLES_COLLECTION_ID),
+                REQUESTS_TIME_OUT_SECONDS,
+                ATTEMPTS_NUMBER,
+                ct,
+                LambdaPaginatedResponseHelper.GetPageSizeParam(1),
```

Comment:
Is it ever paginated or you can delete these parameters?

Reply:
Yep, we can delete them. Done!